pkg/querier/frontend/frontend.go

```diff
@@ -139,7 +143,20 @@ func (f *Frontend) Handler() http.Handler {
 }
 
 func (f *Frontend) handle(w http.ResponseWriter, r *http.Request) {
+	startTime := time.Now()
 	resp, err := f.roundTripper.RoundTrip(r)
+	queryResponseTime := time.Now().Sub(startTime)
+
+	userID, err := user.ExtractOrgID(r.Context())
```

Comment:
Would be better to move it at the beginning of the function, to avoid running the query and get the results if then we trash them because of this error

Reply:
done.